### PR TITLE
introduce miniz submodule

### PR DESCRIPTION
**Pull request by zhaozg**

### CODE REVIEW

**Comment by zhaozg** [comment]
If https://github.com/richgel999/miniz/pull/315 be accept, we can change miniz repo to https://github.com/richgel999/miniz

**Comment by zhaozg** [comment]
https://github.com/richgel999/miniz/pull/315 has been merged, miniz repo back to  https://github.com/richgel999/miniz

File: CMakeLists.txt [hunk 1]
```diff
@@ -132,8 +132,16 @@ if (WITH_LUA_ENGINE STREQUAL Lua)
 endif ()
 
 include(LuaAddExecutable)
+
+# TODO: define MINIZ_NO_STDIO for small size
```
Comment (squeek502) [inline CMakeLists.txt:136]: Is this important to do before merging?

Comment (zhaozg) [reply CMakeLists.txt:136]: No, I hope to make PR to miniz to support MINIZ_NO_STDIO build with cmake later, miniz cmake without MINIZ_NO_STDIO options now. So let's merge, and fix this TODO after miniz with this feature.

Comment (zhaozg) [reply CMakeLists.txt:136]: The miniz PR is https://github.com/richgel999/miniz/pull/316

**Review by squeek502 — APPROVED** [review]

**Pull request merged**